db_stress_tool/db_stress_test_base.cc

```diff
@@ -2327,11 +2327,13 @@ void StressTest::TestCompactFiles(ThreadState* thread,
                       << compact_options.output_file_size_limit;
       auto s = db_->CompactFiles(compact_options, column_family, input_files,
                                  static_cast<int>(output_level));
-      if (!s.ok() && !s.IsManualCompactionPaused() && !s.IsAborted() &&
-          !std::strstr(s.getState(), "injected") &&
-          !(s.IsInvalidArgument() &&
-            std::strstr(s.getState(), "Specified compaction input file") &&
-            std::strstr(s.getState(), "does not exist in column family"))) {
+      // TOOD (hx235): allow an exact list of tolerable failures under stress
+      // test
+      bool non_ok_status_allowed = s.IsManualCompactionPaused() ||
+                                   std::strstr(s.getState(), "injected") ||
```

Comment:
it seems state can be nullptr when s is ok

Reply:
Fixed